modules/tinymce/src/plugins/link/test/ts/browser/NormalizeLinkTest.ts

```diff
@@ -0,0 +1,120 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { Optional } from '@ephox/katamari';
+import { KAssert } from '@ephox/katamari-assertions';
+import { TinyHooks } from '@ephox/mcagar';
+import { Hierarchy, SugarElement } from '@ephox/sugar';
+import Editor from 'tinymce/core/api/Editor';
+import { getSelectedLink } from 'tinymce/plugins/link/core/NormalizeLink';
+import Theme from 'tinymce/themes/silver/Theme';
+interface BaseRange {
+  startPath: number[];
+  startOffset: number;
+  endPath: number[];
+  endOffset: number;
+}
+describe('browser.tinymce.plugins.link.NormalizeLink', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    add_unload_trigger: false,
+    base_url: '/project/tinymce/js/tinymce',
+    indent: false
+  }, [ Theme ], true);
+  const makeRange = (startPath: number[], startOffset: number, endPath: number[], endOffset: number): BaseRange => {
+    return { startPath, startOffset, endPath, endOffset };
+  };
+  const makeCaretRange = (path: number[], offset: number): BaseRange => {
+    return { startPath: path, startOffset: offset, endPath: path, endOffset: offset };
+  };
+  const resolveTestRange = (scope: SugarElement<Node>, range: BaseRange): Range => {
+    const doc = scope.dom.ownerDocument;
+    const rng = doc.createRange();
+    const startContainer = Hierarchy.follow(scope, range.startPath).getOrDie('Failed to resolve start path');
+    const endContainer = Hierarchy.follow(scope, range.endPath).getOrDie('Failed to resolve end path');
+    rng.setStart(startContainer.dom, range.startOffset);
+    rng.setEnd(endContainer.dom, range.endOffset);
+    return rng;
+  };
+  const test = (html: string, range: BaseRange, expectedLink: Optional<number[]>): void => {
+    const editor = hook.editor();
+    const scope = SugarElement.fromDom(editor.getBody());
+    editor.setContent(html);
+    const resolvedRange = resolveTestRange(scope, range);
+    editor.selection.setRng(resolvedRange);
+    const link = getSelectedLink(editor, editor.selection.getNode()).getOrNull();
```

Comment:
Since this is returning an option just bind over `Hierarchy.path` then you wouldn't need the solvedLink part below. So remove getOrNull as well.

Reply:
updated